variables.tf

```diff
@@ -86,6 +86,24 @@ variable "storage" {
   default     = "none"
 }
 
+variable "workloads" {
+  type        = map
+  description = "Workloads to apply on provisioning"
+  default = {
+    ceph_common          = "https://raw.githubusercontent.com/rook/rook/release-1.0/cluster/examples/kubernetes/ceph/common.yaml"
```

Comment:
This seems like an easy way to define the URL for the known services.

Do we need to provide a way to accept new services that we haven't prescribed? 
Should we offer a way to disable these services? 

Could some of these keys be combined, like "cni" for both calico and flannel, if we don't have any other branches in code other than "which URL should we use?".  Doing so would mean that a user needs to localize this map (and maintain it) to change CNIs, so perhaps that is not something we should do.

These are just questions I asked myself, no response needed. Looks good.

Reply:
Definitely agree. I would like to combine some of these (exactly as you suggested), and then a method for users to define additional workloads, but think this should be its own issue, since my desire here was just to kick off that transition away from hardcoding the urls in the template.

Reply:
Noting this for the issue I'll open about it: but for calico, for example, there's two URLs rather than one for flannel, so we'll need to determine if we want to assume nested maps (even if it only contains a single value, etc.); for `cni` this is fairly straight forward (because we're only generating manifests on the fly for things like MetalLB, which we may replace in this project, or the ceph-backing for rook). How to handle the rest is the open question for the other issue/accompanying PR.